components/calendar/calendar.js

```diff
@@ -0,0 +1,480 @@
+		this._shownMonth = this.selectedValue ? this.selectedValue.getMonth() : this._today.getMonth();
+		this._shownYear = this.selectedValue ? this.selectedValue.getFullYear() : this._today.getFullYear();
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+
+		changedProperties.forEach((oldVal, prop) => {
+			if (prop === '_shownMonth') {
+				this._focusDateAddFocus();
+			}
+		});
+	}
+
+	render() {
+		const weekdayHeaders = this._daysOfWeekIndex.map((index) => html`
+			<th role="columnheader" abbr="${this._descriptor.calendar.days.long[index]}">
+				<abbr class="d2l-body-small" title="${this._descriptor.calendar.days.long[index]}">
+					${this._descriptor.calendar.days.short[index]}
+				</abbr>
+			</th>
+		`);
+
+		const dates = getDatesInMonthArray(this._shownMonth, this._shownYear, this._descriptor.calendar.firstDayOfWeek);
+		const dayRows = dates.map((week, index) => {
+			const weekHtml = week.map((day) => {
+				const classes = {
+					'd2l-calendar-date': true,
+					'd2l-calendar-date-first-row': index === 0 ? true : false,
+					'd2l-calendar-date-selected': checkIfDatesEqual(day, this.selectedValue),
+					'd2l-calendar-date-today': checkIfDatesEqual(day, this._today)
+				};
+				const focused = checkIfDatesEqual(day, this._focusDate);
+				return html`
+					<td>
+						<div
+							@click="${this._onDateSelected}"
+							@keydown="${this._onKeyDown}"
+							class=${classMap(classes)}
+							data-date=${day.getDate()}
+							data-month=${day.getMonth()}
+							data-year=${day.getFullYear()}
+							tabindex=${focused ? '0' : '-1'}>
+							${day.getDate()}
+						</div>
+					</td>`;
+			});
+
+			return html`
+				<tr>${weekHtml}</tr>
+			`;
+		});
+		const heading = `${this._descriptor.calendar.months.long[this._shownMonth]} ${this._shownYear}`;
+		return html`
+			<div aria-labelledby="${this._dialogLabelId}" class="d2l-calendar">
+				<table summary="${this.summary}" role="grid">
+					<thead>
+						<tr class="d2l-calendar-title">
+							<td>
+								<d2l-button-icon
+									@click="${this._onPrevMonthButtonClick}"
+									text="${this._computeText(getPrevMonth(this._shownMonth))}"
+									icon="tier1:chevron-left">
+								</d2l-button-icon>
+							</td>
+							<th colspan="5">
+								<h2 class="d2l-heading-4" aria-live="polite" id="${this._dialogLabelId}">${heading}</h2>
+							</th>
+							<td>
+								<d2l-button-icon
+									@click="${this._onNextMonthButtonClick}"
+									text="${this._computeText(getNextMonth(this._shownMonth))}"
+									icon="tier1:chevron-right">
+								</d2l-button-icon>
+							</td>
+						</tr>
+					</thead>
+					<tbody>
+						<tr>
+							${weekdayHeaders}
+						</tr>
+						${dayRows}
+					</tbody>
+				</table>
+				<slot></slot>
+			</div>
+		`;
+	}
+
+	_computeText(month) {
+		return this.localize('show', {month: this._descriptor.calendar.months.long[month]});
+	}
+
+	_focusDateAddFocus() {
```

Comment:
Another option to all this code that's removing/adding tabindex to the currently focused date would be to move `_focusDate` to be a private property being tracked by Lit. That way whenever it changes, it would call render again which already handles setting the `tabindex` on the focussed date.

Reply:
This seems more ideal. `this._focusDate.setDate` doesn't actually trigger a re-render which is a bit of a pain, but it works if it's changed with something like `this._focusDate = new Date...`

Reply:
Cool, I think it'll end up with a more logical flow as well. It'll also prevent situations where the focus moves and updates tabindexes but then a re-render happens right away anyway.